modules/lang-painless/src/main/java/org/elasticsearch/painless/ScriptImpl.java

```diff
@@ -119,8 +119,9 @@ public void setNextAggregationValue(Object value) {
     public Object run() {
         try {
             return executable.execute(variables, scorer, doc, aggregationValue);
-        } catch (PainlessError | BootstrapMethodError | IllegalAccessError | Exception t) {
-            throw convertToScriptException(t);
+        // Note that it is safe to catch any of the following errors since Painless is stateless.
+        } catch (PainlessError | BootstrapMethodError | OutOfMemoryError | StackOverflowError | Exception e) {
+            throw convertToScriptException(e);
         }
```

Comment:
thanks for killing this IllegalAccessError: i added this on accident when debugging long ago!